hw1_mzw7af_vl4kz.py

```diff
@@ -57,6 +57,30 @@ def main():
     facts = {}
     rules = []
 
+    for line in sys.stdin:
+        argArray = line.strip().split(" ")
 
+        if argArray[0].lower() == "teach":
+            if argArray[3] == "=":
+                stringVar = ' '.join(argArray).split(" = ")[1]
+                teachVar(argArray[1], argArray[2], stringVar)
+                #print(argArray[1] + " " + argArray[2] + " " + stringVar)
+            elif argArray[2] == "=":
+                teachRootVar(argArray[1], argArray[len(argArray)-1])
+                #print(argArray[1] + " " + argArray[len(argArray)-1])
+            else:
+                expression = ' '.join(argArray[1:]).split(" -> ")
+                teachRule(expression[0], expression[len(expression)-1])
+                #print(expression[0] + " " + expression[len(expression)-1])
+        elif argArray[0].lower() == "list":
+            listInst()
+        elif argArray[0].lower() == "learn":
+            learn()
+        elif argArray[0].lower() == "query":
+            query(argArray[len(argArray)-1])
+            #print(argArray[len(argArray)-1])
+        elif argArray[0].lower() == "why":
+            why(argArray[len(argArray)-1])
```

Comment:
len - 1 all over the place sadness 

Also remove comments once youre done testing plz